### ISSUE TITLE
Dialog `aria-disabled` attribute isn't changed when dialogs are opened

### ISSUE THREAD

**Issue by strackoverflow** [primer/view_components#1988]
There appears to be an accessibility issue with `modal-dialog`. The dialog gets an `aria-disabled="true"` attribute [when it is rendered](https://github.com/primer/view_components/blob/d100a251ec7dad6a09ce177a2495fff0ac23c2f2/app/components/primer/alpha/dialog.rb#L151-L154), but when the dialog is opened, the attribute remains `true`. 

The effect of this is that screen readers will read every element inside of the semantically "disabled" dialog as `unavailable` (example, the close button will be announced as "Close button, unavailable"). This is repeated for every element in the dialog.

The `modal-dialog` custom element should be updated to toggle this attribute when the modal is opened and closed so that screen reader users can more easily interact with dialogs.

**Comment by benjiallen** [comment]
Quick demo of what happens when using a screen reader and opening the dialog within the global navigation. All elements within the dialog are announced as unavailable.

https://user-images.githubusercontent.com/16546334/236009949-a3e3f852-2ece-4414-8074-e21eff4396c8.mp4

**Comment by lesliecdubs** [comment]
👋 @keithamus is this something you might be able to take a look at this week given your familiarity with the dialog component?
